relationships.md

```diff
@@ -0,0 +1,280 @@
+
+## Single Relationship
+**/api/core/relationships/<:id>**
+
+A sample can be found at https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/relationships/530
+The leftId and rightId parameter in JSON may still be present, but they will be removed soon (requires other PRs to be merged first)
+
+```json
+{
+  "id": 530,
+  "relationshipTypeId": 0,
+  "leftPlace": 1,
+  "rightPlace": 1,
+  "type": "relationship",
+  "_links": {
+    "relationshipType": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/relationshiptypes/1"
+    },
+    "self": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/relationships/530"
+    },
+    "leftItem": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/items/e98b0f27-5c19-49a0-960d-eb6ad5287067"
+    },
+    "rightItem": {
+      "href": "https://dspace7-entities.atmire.com/rest/api/core/items/0ffbee3f-e7ea-42bc-92fe-2fbef1a52c0f"
+    }
+  },
+  "_embedded": {
+    "relationshipType": {
+      "id": 1,
+      "leftLabel": "isAuthorOfPublication",
+      "rightLabel": "isPublicationOfAuthor",
+      "leftMinCardinality": 0,
+      "leftMaxCardinality": 2147483647,
+      "rightMinCardinality": 0,
+      "rightMaxCardinality": 20000,
+      "type": "relationshiptype",
+      "_links": {
+        "leftType": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1"
+        },
+        "rightType": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2"
+        },
+        "self": {
+          "href": "https://dspace7-entities.atmire.com/rest/api/core/relationshiptypes/1"
+        }
+      },
+      "_embedded": {
+        "leftType": {
+          "id": 1,
+          "label": "Publication",
+          "type": "entitytype",
+          "_links": {
+            "self": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1"
+            },
+            "relationshiptypes": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/1/relationshiptypes"
+            }
+          }
+        },
+        "rightType": {
+          "id": 2,
+          "label": "Person",
+          "type": "entitytype",
+          "_links": {
+            "self": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2"
+            },
+            "relationshiptypes": {
+              "href": "https://dspace7-entities.atmire.com/rest/api/core/entitytypes/2/relationshiptypes"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The [relationship type](relationshiptypes.md) is embedded
+The 2 items are included as HAL links but are not embedded
+
+## Relationships per Relationship type
+**/api/core/relationships/search/byName?relationshipname=<:relationshipname>**
```

Comment:
A tiny nitpicky thing, I'm noticing we call this "relationshipname" and "byName" here.  But, in the JSON returned these are called *labels*.  It's a bit confusing to be calling these both names and labels.  Should this endpoint be `/api/core/relationships/search/byLabel?label=<:relationshiplabel>`?  Or is there a plan to change these into "name" fields elsewhere?

Reply:
You're right, we do call them labels